### ISSUE TITLE
[BUG]Failed to submit orders to Kucoin

### ISSUE THREAD

**Issue by RC-13** [CoinAlpha/hummingbot#2162]
**Describe the bug**
After updating to the latest development commit, orders are not getting submitted to KuCoin.
```
2020-08-06 01:51:51,002 - 1707 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (KCS-ETH) Creating 1 bid orders at (Size, Price): ['5 KCS, 0.002261 ETH']
2020-08-06 01:51:51,004 - 1707 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (KCS-ETH) Creating 1 ask orders at (Size, Price): ['5 KCS, 0.0023298 ETH']
2020-08-06 01:51:51,283 - 1707 - hummingbot.market.kucoin.kucoin_market - NETWORK - Error submitting buy limit_maker order to Kucoin for 5.0000 KCS-ETH 0.0022610.
Traceback (most recent call last):
  File "hummingbot/market/kucoin/kucoin_market.pyx", line 707, in hummingbot.market.kucoin.kucoin_market.KucoinMarket.execute_buy
    exchange_order_id = await self.place_order(order_id, trading_pair, decimal_amount, True, order_type, decimal_price)
  File "hummingbot/market/kucoin/kucoin_market.pyx", line 684, in place_order
    return str(exchange_order_id["data"]["orderId"])
KeyError: 'data'
2020-08-06 01:51:51,286 - 1707 - hummingbot.market.kucoin.kucoin_market - WARNING - Failed to submit buy order to Kucoin. Check API key and network connection.
2020-08-06 01:51:51,366 - 1707 - hummingbot.market.kucoin.kucoin_market - NETWORK - Error submitting sell limit_maker order to Kucoin for 5.0000 KCS-ETH 0.0023298.
Traceback (most recent call last):
  File "hummingbot/market/kucoin/kucoin_market.pyx", line 777, in hummingbot.market.kucoin.kucoin_market.KucoinMarket.execute_sell
    exchange_order_id = await self.place_order(order_id, trading_pair, decimal_amount, False, order_type, decimal_price)
  File "hummingbot/market/kucoin/kucoin_market.pyx", line 684, in place_order
    return str(exchange_order_id["data"]["orderId"])
KeyError: 'data'
2020-08-06 01:51:51,367 - 1707 - hummingbot.market.kucoin.kucoin_market - WARNING - Failed to submit sell order to Kucoin. Check API key and network connection.
```

**Steps To Reproduce**
// A concise description of the **steps to reproduce the buggy behavior**:

Here are the steps to reproduce the issue (see attachments in the section below):
1. Update to latest development commit
2. Create a pure market making strategy
3. Start and wait for few minutes until the orders are created.

**Screenshots**
Source build:
![dev-source](https://user-images.githubusercontent.com/57189990/89451065-330c6680-d78e-11ea-9ccd-965e6088fdc3.JPG)
Docker build:
![dev](https://user-images.githubusercontent.com/57189990/89451069-36075700-d78e-11ea-85da-c05529bfc4b6.JPG)

Master branch: No issue
![master](https://user-images.githubusercontent.com/57189990/89451090-3c95ce80-d78e-11ea-87cc-df08a0becddb.JPG)

**Release version**
0.30.0
```
branch: "development",
commit: "47ebb83a50fd33e1a6902ee6775fe8f858ce0fa5"
date: "2020-08-05T15:22:10Z",
```

**Attachments**
[logs_KuCoin.log](https://github.com/CoinAlpha/hummingbot/files/5030494/logs_KuCoin.log)

*WARNING: Do NOT publish any exchange API keys or your wallet's private key. Whoever has access to them may steal your assets!*

**Optional: your discord username:**

**Comment by TheHolyRoger** [comment]
That's very strange... https://github.com/CoinAlpha/hummingbot/compare/master...development#diff-4dd788b203c9c7f8e580f6113d1da1ecR684

**Comment by PtrckM** [comment]
confirmed fix on 0.30.0

![image](https://user-images.githubusercontent.com/54516858/90132186-ed5c2900-dd9f-11ea-8060-19453eaf5dd0.png)
![image](https://user-images.githubusercontent.com/54516858/90132198-f4833700-dd9f-11ea-99f9-3b8345c63c82.png)

**Comment by masterpi227** [comment]
this error is still occuring in v0.38

**Comment by glenntomas** [comment]
Hi @masterpi227, this bug ticket is fixed on #2186. Can you please send a screenshot of your whole hummingbot terminal after running `config` command and also if there's an error on the log pane? Please also send your log file so we could check what's causing the issue your encountering. Make sure to check your Kucoin main account and trading account, your assets should be on your trading account. Thank you!

**Comment by masterpi227** [comment]
Traceback (most recent call last):
  File "hummingbot/connector/exchange/kucoin/kucoin_exchange.pyx", line 697, in hummingbot.connector.exchange.kucoin.kucoin_exchange.KucoinExchange.execute_buy
    exchange_order_id = await self.place_order(order_id, trading_pair, decimal_amount, True, order_type,
  File "hummingbot/connector/exchange/kucoin/kucoin_exchange.pyx", line 674, in place_order
    return str(exchange_order_id["data"]["orderId"])
KeyError: 'data'

I could not find any catch for this type of Error or some preconditions in the code.

Further, I identified the problem. There is a critical error in arbitrage strategy logic. see down below the code from arbitrage.pyx, buy_price and sell_price are mixed up. This leads to wrong trade processing. in my case, kucoin balance was too low to procees wrong amount*price, that led to the error, because kucoin rejected the trade.

`best_amount, best_profitability, buy_price, sell_price = self.c_find_best_profitable_amount(            buy_market_trading_pair_tuple, sell_market_trading_pair_tuple        )`

**Comment by glenntomas** [comment]
Hi @masterpi227, I think this is the fix to your concern #3158 but it's not yet merged. You can monitor the PR for updates. Thank you!
